pkgs/build-support/setup-hooks/strip.sh

```diff
@@ -49,8 +49,14 @@ stripDirs() {
     local ranlibCmd="$2"
     local paths="$3"
     local stripFlags="$4"
+    local excludeFlags=()
     local pathsNew=
 
+    local ext
+    for ext in ${stripExcludeExtensions:-}; do
+      excludeFlags+=(-a '!' -name "*.$ext")
```

Comment:
I wonder if we could make this more general?  e.g. in Linux we want to avoid stripping certain glob patterns:

https://github.com/NixOS/nixpkgs/blob/d59c85209d3d43b9867694b5c747db32ce977c7e/pkgs/os-specific/linux/kernel/manual-config.nix#L376-L379

IIRC bash can do negative globs, so if we had a mechanism for populating `stripDebugList` based on a glob pattern by setting a Nix attribute, that would solve both cases.

Reply:
(I understand that you found it difficult to get working with the glob expressed in Nix, but I think we should take the time to make sure that the API is the best it can be, since it's difficult to change later.  If you want to decouple that from fixing the Rust stuff, you can probably take inspiration from the code above?)

Reply:
I agree it's worth attempting to get something proper that doesn't fall over if you include a `*`.

But the example you give would require path globs. I was only intending to support filename globs, so I still don't know if the two are compatible. If we make it a full path glob, then the user has to know upfront what (recursive) path patterns there would be.

Reply:
We could use bash's globstar option, which would allow extension globbing with a pattern like "**/*.so".